app/src/main/java/org/mozilla/fenix/browser/BrowserStore.kt

```diff
@@ -0,0 +1,26 @@
+package org.mozilla.fenix.browser
+
+import mozilla.components.browser.state.action.BrowserAction
+import mozilla.components.browser.state.state.BrowserState
+import mozilla.components.browser.state.store.BrowserStore
+import mozilla.components.feature.media.ext.findActiveMediaTab
+import mozilla.components.lib.state.Store
+
+fun BrowserStore.waitForPIPToClose(
+    block: () -> Unit
+) {
+
+    if (state.findActiveMediaTab() == null || !state.findActiveMediaTab()?.mediaSessionState?.fullscreen!!) {
```

Comment:
Thank you for reviewing! I'll update the code.

Reply:
@Mugurell , if we go to search fragment immediately the screen orientation will still be in landscape mode. 
When the BrowserFragment receives  pipmode as false, FullScreenFeature.onBackPressed() is invoked which triggers the full screen exit event which finally reaches MediaSessionFullscreenFeature. When MediaSessionFullscreenFeature finds out that the fullscreen is false only then the activity orientation is set to portrait. Will explore more..

Reply:
Maybe the intent processor can also set sensor orientation?
It might be that we need all the functionality to exit fullscreen here.

Reply:
Sure. We can do it that way

Reply:
@Mugurell , I've updated the changes. But the screen first goes to the landscape mode and then to the portrait mode, since the activity is in landscape while it's in PIP mode. 


https://user-images.githubusercontent.com/12825812/166416361-9af290a2-1686-4107-8f20-4ad44f0e740c.mp4



Maybe, we could just set the activity's orientation to ActivityInfo.SCREEN_ORIENTATION_UNSPECIFIED in MediaSessionFullscreenFeature, when the activity is in PIP mode. Is this a good way to solve the problem? If so I'll look deeper into the MediaSessionFullscreenFeature class since it's used by several classes.

```
private fun processFullscreen(sessionStates: List<SessionState>) {
        /* there should only be one fullscreen session */
        val activeState = sessionStates.firstOrNull()
        if (activeState == null || activeState.mediaSessionState?.fullscreen != true) {
            activity.requestedOrientation = ActivityInfo.SCREEN_ORIENTATION_USER
            return
        }

        when (activeState.mediaSessionState?.elementMetadata?.portrait) {
            true ->
                activity.requestedOrientation =
                    ActivityInfo.SCREEN_ORIENTATION_USER_PORTRAIT
            false ->
                if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.N && activity.isInPictureInPictureMode) {
                    activity.requestedOrientation = ActivityInfo.SCREEN_ORIENTATION_UNSPECIFIED
                }else{
                    activity.requestedOrientation =
                        ActivityInfo.SCREEN_ORIENTATION_SENSOR_LANDSCAPE
                }

            else -> activity.requestedOrientation = ActivityInfo.SCREEN_ORIENTATION_USER
        }
    }
```
When we search from search widget

https://user-images.githubusercontent.com/12825812/166416514-28fa3acd-a248-40ee-86d8-123052d96f73.mp4

Also when we enter from PIP (from home)

https://user-images.githubusercontent.com/12825812/166416622-d61cc357-6aeb-4b03-84b4-7839b853a06c.mp4